### PR TITLE
Shoot conditions `lastUpdateTime` is no longer updated continuously

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area scalability robustness
/kind enhancement
/priority normal

**What this PR does / why we need it**:
The shoot's `.status.conditions[].lastUpdateTime` is no longer continuously updated to prevent spamming etcd with too many unnecessary updates.

**Special notes for your reviewer**:
If the gardenlet does not renew its lease anymore then the GCM will, after a certain threshold, set the shoot conditions to `Unknown` to indicate that the displayed information is most likely outdated.

**Release note**:

```improvement operator
The shoot's `.status.conditions[].lastUpdateTime` is no longer continuously updated. Note that if the gardenlet does not renew its lease anymore then the gardener-controller-manager will, after a certain threshold, set the `Shoot`'s conditions to `Unknown` to indicate that the displayed information is most likely outdated.
```

### CODE REVIEW

**Comment by vlerenc** [comment]
Thanks @rfranzke. When we merge it/deploy it to dev, we should check whether it indeed behaves as expected and the no longer updated `.status.conditions[].lastUpdateTime` will not cause trouble in the components.

WDYT if we no longer update `.status.conditions[].lastUpdateTime` for shoots:
* @grolu for the dashboard
* @wyb1 @mliepold for the availability monitoring

**Comment by petersutter** [comment]
we have no logic that depends on `.status.conditions[].lastUpdateTime`. 

This would be the only effect in case it is not set (Last Updated will show nothing), however we have a PR ready to remove this field https://github.com/gardener/dashboard/pull/810

Screenshot 2020-09-24 at 15 55 52

**Comment by mvladev** [comment]
Btw there is already a condition builder which checks if condition really needs to be updated e.g. https://github.com/gardener/gardener/blob/2fb3b07c5c741b432889ce155a89a82cd88532b1/pkg/gardenlet/controller/seed/seed_lease_control.go#L101-L120

File: pkg/apis/core/v1beta1/helper/helper.go [hunk 1]
```diff
@@ -92,13 +76,16 @@ func UpdatedCondition(condition gardencorev1beta1.Condition, status gardencorev1
...
-		LastUpdateTime:     Now(),
+		LastUpdateTime:     condition.LastUpdateTime,
 		Codes:              codes,
 	}
 
 	if condition.Status != status {
-		newCondition.LastTransitionTime = Now()
+		now := Now()
+		newCondition.LastTransitionTime = now
+		newCondition.LastUpdateTime = now
```
Comment (mvladev) [inline pkg/apis/core/v1beta1/helper/helper.go]: The LastUpdate and LastTransitionTime should be updated only under specific conditions e.g. update to status / reason or message:

https://github.com/gardener/gardener/blob/24334336a6f069b00ad5c7b839fac980aa9ea5f8/pkg/apis/core/v1beta1/helper/condition_builder.go#L100-L155

**Review by mvladev — CHANGES_REQUESTED** [review]

**Review by amshuman-kr — DISMISSED** [review]
I am not aware of all the details of how these conditions and constraints are used. But looks good in principle :-)

**Review by petersutter — DISMISSED** [review]

**Comment by rfranzke** [comment]
> Btw there is already a condition builder which checks if condition really needs to be updated e.g.

Yes, I know, actually I started using it, but then I faced that the `botanist/health_check.go` would be adapted as well, resulting in a larger refactoring. I have a WIP branch for it and will continue separately.

File: pkg/apis/core/v1beta1/helper/helper.go (hunk 1 above)
Comment (rfranzke) [reply pkg/apis/core/v1beta1/helper/helper.go]: done

File: pkg/utils/kubernetes/health/health.go [hunk 2]
```diff
@@ -342,10 +342,10 @@ func CheckAPIServerAvailability(ctx context.Context, condition gardencorev1beta1
 		} else {
 			body = string(bodyRaw)
 		}
-		message := fmt.Sprintf("API server /healthz endpoint endpoint check returned a non ok status code %d. %s (%s)", statusCode, responseDurationText, body)
+		message := fmt.Sprintf("API server /healthz endpoint endpoint check returned a non ok status code %d. (%s)", statusCode, body)
```
Comment (vpnachev) [inline pkg/utils/kubernetes/health/health.go]:
```suggestion
		message := fmt.Sprintf("API server /healthz endpoint check returned a non ok status code %d. (%s)", statusCode, body)
```

**Review by vpnachev — DISMISSED** [review]
/lgtm

**Review by vpnachev — APPROVED** [review]
/lgtm

**Review by mvladev — APPROVED** [review]
/lgtm

**Pull request merged**